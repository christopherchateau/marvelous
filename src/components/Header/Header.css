.Header {
  background: linear-gradient(to right, red, rgb(255, 83, 112));
  background-color: red;
  display: flex;
  grid-area: header;
  height: 200px;
  justify-content: space-around;
  text-align: center;
}

.main-logo {
  background: url("../../images/wolverine-bg.jpg") center top no-repeat;
  background-size: 710px;
  bottom: 20px;
  color: white;
  font-family: "Anton", sans-serif;
  font-size: 150px;
  position: relative;
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
}

.explore-marvel-text {
  bottom: 57px;
  color: white;
  font-family: "Roboto", sans-serif;
  font-size: 26px;
  position: relative;
}

.spider-man-pic,
.wolverine-pic {
  height: 190px;
  width: 190px;
}

.spider-man-pic {
  position: relative;
  top: 6px;
}

@media screen and (max-width: 1060px) {
  .spider-man-pic,
  .wolverine-pic {
    display: none;
  }
}

@media screen and (max-width: 720px) {
  .main-logo {
    font-size: 120px;
    padding: 20px 0;
  }

  .explore-marvel-text {
    bottom: 60px;
    font-size: 20px;
  }
}

@media screen and (max-width: 560px) {
  .Header {
    height: 140px;
  }

  .main-logo {
    font-size: 90px;
  }

  .explore-marvel-text {
    font-size: 14px;
  }
}

@media screen and (max-width: 420px) {
  .Header {
    height: 100px;
  }

  .main-logo {
    font-size: 70px;
    padding: 10px 0;
  }

  .explore-marvel-text {
    bottom: 44px;
    font-size: 10px;
  }
}
