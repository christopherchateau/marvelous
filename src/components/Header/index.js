import React from "react";
import spiderMan from "../../images/spider-man.png";
import wolverine from "../../images/wolverine.png";
import "./Header.css";

const Header = () => {
  return (
    <header className="Header">
      <img className="wolverine-pic" alt="wolverine" src={wolverine} />
      <article>
        <h1 className="main-logo">MARVELOUS</h1>
        <h4 className="explore-marvel-text">
          Explore the Marvel Universe one character at at time.
        </h4>
      </article>
      <img className="spider-man-pic" alt="spider-man" src={spiderMan} />
    </header>
  );
};

export default Header;
