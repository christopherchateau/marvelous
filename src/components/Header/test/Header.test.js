import React from "react";
import { shallow } from "enzyme";
import Header from "../index";

describe("Header", () => {
  const wrapper = shallow(<Header />);
  it("should exist", () => {
    expect(wrapper).toBeDefined;
  });

  it("should render like snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
