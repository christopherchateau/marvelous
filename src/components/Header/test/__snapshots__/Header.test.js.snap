// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render like snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt="wolverine"
          className="wolverine-pic"
          src="wolverine.png"
        />,
        <article>
          <h1
            className="main-logo"
          >
            MARVELOUS
          </h1>
          <h4
            className="explore-marvel-text"
          >
            Explore the Marvel Universe one character at at time.
          </h4>
        </article>,
        <img
          alt="spider-man"
          className="spider-man-pic"
          src="spider-man.png"
        />,
      ],
      "className": "Header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "wolverine",
          "className": "wolverine-pic",
          "src": "wolverine.png",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1
              className="main-logo"
            >
              MARVELOUS
            </h1>,
            <h4
              className="explore-marvel-text"
            >
              Explore the Marvel Universe one character at at time.
            </h4>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "MARVELOUS",
              "className": "main-logo",
            },
            "ref": null,
            "rendered": "MARVELOUS",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Explore the Marvel Universe one character at at time.",
              "className": "explore-marvel-text",
            },
            "ref": null,
            "rendered": "Explore the Marvel Universe one character at at time.",
            "type": "h4",
          },
        ],
        "type": "article",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "spider-man",
          "className": "spider-man-pic",
          "src": "spider-man.png",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
    ],
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="wolverine"
            className="wolverine-pic"
            src="wolverine.png"
          />,
          <article>
            <h1
              className="main-logo"
            >
              MARVELOUS
            </h1>
            <h4
              className="explore-marvel-text"
            >
              Explore the Marvel Universe one character at at time.
            </h4>
          </article>,
          <img
            alt="spider-man"
            className="spider-man-pic"
            src="spider-man.png"
          />,
        ],
        "className": "Header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "wolverine",
            "className": "wolverine-pic",
            "src": "wolverine.png",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1
                className="main-logo"
              >
                MARVELOUS
              </h1>,
              <h4
                className="explore-marvel-text"
              >
                Explore the Marvel Universe one character at at time.
              </h4>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "MARVELOUS",
                "className": "main-logo",
              },
              "ref": null,
              "rendered": "MARVELOUS",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Explore the Marvel Universe one character at at time.",
                "className": "explore-marvel-text",
              },
              "ref": null,
              "rendered": "Explore the Marvel Universe one character at at time.",
              "type": "h4",
            },
          ],
          "type": "article",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "spider-man",
            "className": "spider-man-pic",
            "src": "spider-man.png",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
      ],
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
